nemo/collections/nlp/models/language_modeling/megatron_bert_model.py

```diff
@@ -139,45 +139,94 @@ def __init__(self, cfg: DictConfig, trainer: Trainer):
             self._nsys_profile_start_step *= grad_accum_steps
             self._nsys_profile_end_step *= grad_accum_steps
 
+    def _wrap_model_for_O2(self):
```

Comment:
do we need to repeat this code for each model? I think we have a simliar method for gpt already maybe move to base class and reuse?

Reply:
I left this one as such since, it feels like you. need to add a lot of conditionals to check what model type it is in the base clasee (GPT and bert have different stuff in most parts of the code). So probabaly leaving it now. Maybe if other models find more commanlity with GPT or something, we can put it in base class and I can just override it in bert model.